client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -912,7 +912,14 @@ private static Exception extractAndWrapCause(Exception exception) {
             return e;
         }
         if (exception instanceof SSLHandshakeException) {
-            SSLHandshakeException e = new SSLHandshakeException(exception.getMessage());
+            SSLHandshakeException e;
+            if (exception.getMessage().contains("SunCertPathBuilderException")) {
```

Comment:
Any reason not to always include this? Troubleshooting instructions could contain all kinds of things.

Reply:
Are you suggesting having a troubleshooting page in documentation for common errors? 

Current thinking was to point the most commonly missed client side certificate setup (symptom `SunCertPathBuilderException`) -- to the [documentation](https://opensearch.org/docs/latest/clients/java-rest-high-level/).

Reply:
I was thinking any `SSLHandshakeException` would be fair game and more future-proof. 

Don't have strong opinions. However if we specialize the error based on what's in the message we need tests for that so that we don't accidentally remove it.

Reply:
>I was thinking any SSLHandshakeException would be fair game and more future-proof.

Got it. We can keep it at SSLHandshakeException itself. I like the idea of keeping it future-proof and only adding more troubleshooting help to documentation for more scenarios.

Reply:
So I should remove the if statement so the troubleshooting message applies to all SSLHandshakeExceptions?

Reply:
yes